conda_build/main_build.py

```diff
@@ -308,6 +309,13 @@ def execute(args, parser):
                 raise RuntimeError("%s must be major.minor, not %s" %
                     (conda_version[lang], version))
 
+    # Using --python, --numpy etc. is equivalent to using CONDA_PY, CONDA_NPY, etc.
+    # Auto-set those env variables
+    for var in conda_version.values():
+        if getattr(config, var):
```

Comment:
Just so I'm clear on this: you are effectively giving priority to any setting passed in with command line stuff, like "--python" over the environment variables, right?  I'm ok with that, just want to make sure I understand.